config/rbac/kustomization.yaml

```diff
@@ -12,7 +12,7 @@ resources:
 # Comment the following 4 lines if you want to disable
 # the auth proxy (https://github.com/brancz/kube-rbac-proxy)
 # which protects your /metrics endpoint.
-#- auth_proxy_service.yaml
-#- auth_proxy_role.yaml
-#- auth_proxy_role_binding.yaml
-#- auth_proxy_client_clusterrole.yaml
+- auth_proxy_service.yaml
+- auth_proxy_role.yaml
+- auth_proxy_role_binding.yaml
+- auth_proxy_client_clusterrole.yaml
```

Comment:
same question, are those for CI or for production?

Reply:
`auth_proxy_role.yaml `and `auth_proxy_role_binding.yaml` are mandatory for the sidecar operation. Meaning these two needed for CI (for curl tests) but also for production.
However, `auth_proxy_service.yaml` is needed for production usage, (it is part of the infrastructure to allow Prometheus to scrape metrics)

Reply:
However, the `auth_proxy_client_clusterrole.yaml` is not being used at all

Reply:
ok, so let's not add `auth_proxy_client_clusterrole.yaml` or unnecessary stuff in general

Reply:
> `auth_proxy_role.yaml, `auth_proxy_service.yaml`and`auth_proxy_role_binding.yaml` are mandatory for the sidecar operation. Meaning these three needed for CI (for curl tests) but also for production. without them sidecar won't be ready. The service deployment is mandatory as its annotation responsible for creating a secret consumed as a volume by the sidecar.

Could you elaborate about "not be ready"? I'd expect the sidecars to be up and running, but not be accessible by anyone in the cluster without these RBAC rules.

Reply:
If the service for example won't be created, the secret that we mount as a volume won't be created. Therefore, this would result with an error and the sidecar state won't be ready.

Reply:
ok, thanks. This is a possible problem for the backports, because makes them more invasive than expected.